src/composer.rs

```diff
@@ -0,0 +1,1052 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+//! PLONK turbo composer definitions
+
+use alloc::vec::Vec;
+use core::cmp;
+use core::ops::Index;
+
+use dusk_bls12_381::BlsScalar;
+use dusk_bytes::Serializable;
+use dusk_jubjub::{JubJubAffine, JubJubExtended, JubJubScalar};
+
+use crate::bit_iterator::BitIterator8;
+use crate::constraint_system::ecc::WnafRound;
+use crate::constraint_system::{
+    Constraint, Selector, WiredWitness, Witness, WitnessPoint,
+};
+use crate::error::Error;
+use crate::runtime::{Runtime, RuntimeEvent};
+
+mod builder;
+mod circuit;
+mod compiler;
+mod polynomial;
+mod prover;
+mod verifier;
+
+pub use builder::Builder;
+pub use circuit::Circuit;
+pub use compiler::Compiler;
+pub use polynomial::Polynomial;
+pub use prover::Prover;
+pub use verifier::Verifier;
+
+/// Circuit builder tool
+pub trait Composer: Sized + Index<Witness, Output = BlsScalar> {
+    /// Zero representation inside the constraint system.
+    ///
+    /// A turbo composer expects the first witness to be always present and to
+    /// be zero.
+    const ZERO: Witness = Witness::new(0);
+
+    /// `One` representation inside the constraint system.
+    ///
+    /// A turbo composer expects the 2nd witness to be always present and to
+    /// be one.
+    const ONE: Witness = Witness::new(1);
+
+    /// Identity point representation inside the constraint system
+    const IDENTITY: WitnessPoint = WitnessPoint::new(Self::ZERO, Self::ONE);
+
+    /// Create an empty constraint system.
+    ///
+    /// This shouldn't be used directly; instead, use [`Self::initialized`]
+    #[deprecated(
+        since = "13.0",
+        note = "this function is meant for internal use. call `initialized` instead"
+    )]
```

Comment:
Its a way to give a warning for users that calls this function directly. Unfortunately, Rust doesn't allow us to make private scope for traits since they are enforced to be public.

We could alternatively just leave this implementation open for the user, but I think its better to give a warning if they attempt to call that.